src/Infrastructure.EntityFramework/Repositories/GroupRepository.cs

```diff
@@ -84,6 +84,45 @@ public async Task<ICollection<Core.Entities.Group>> GetManyByOrganizationIdAsync
         }
     }
 
+    public async Task<ICollection<Tuple<Core.Entities.Group, ICollection<SelectionReadOnly>>>>
+        GetManyWithCollectionsByOrganizationIdAsync(Guid organizationId)
+    {
+        var groups = await GetManyByOrganizationIdAsync(organizationId);
+        using (var scope = ServiceScopeFactory.CreateScope())
+        {
+            var dbContext = GetDatabaseContext(scope);
+            var query = await (
+                from cg in dbContext.CollectionGroups
+                where cg.Group.OrganizationId == organizationId
+                select cg).ToListAsync();
+
+            return groups.Select(group =>
+                new Tuple<Core.Entities.Group, ICollection<SelectionReadOnly>>(
+                    group,
+                    query.Select(c => new SelectionReadOnly
+                    {
+                        Id = c.CollectionId,
+                        ReadOnly = c.ReadOnly,
+                        HidePasswords = c.HidePasswords,
+                    }).ToList()
```

Comment:
I don't think this is grouping the collections by group, it's just associating every collection with every group.

Reply:
Whoops, good catch! Fixed now.